rollup/main/default/classes/RollupCalculator.cls

```diff
@@ -681,10 +681,11 @@ public without sharing abstract class RollupCalculator {
     protected override void setReturnValue() {
       String trimmedDelimiter = this.concatDelimiter.trim();
       String possibleReturnValue = this.stringVal.normalizeSpace();
-      String withoutEndingComma = possibleReturnValue.endsWith(trimmedDelimiter)
-        ? possibleReturnValue.substringBeforeLast(trimmedDelimiter)
-        : possibleReturnValue;
-      this.returnVal = (withoutEndingComma.startsWith(trimmedDelimiter) ? withoutEndingComma.substring(1, withoutEndingComma.length()) : withoutEndingComma)
+      while(possibleReturnValue.endsWith(trimmedDelimiter)) {
```

Comment:
I found while testing the grandparent rollups that it was possible in a delete concat/concat distinct event for the _value_ to to be removed without the delimiter being removed. This is due to the shared code path between update/delete in the `replaceWithDelimiter` function - I'll have to go back and review whether or not those code paths sharing responsibilities makes sense before resolving this; it could be that delete should just always remove both the value and the delimiter. I have good enough test coverage on deletes that returning to this version of the code and trying that should make the matter clear as day!

Reply:
Update - tried the simple approach of reverting this change and updating the `handleDeleteConcat` method to do a replace on both the value _and_ the delimiter, but it didn't work. Since I'm not planning to spend more time on this tonight, I will plan to circle back around to this piece of feedback and clean that method up separately.